### PR TITLE
feat: add warning for outdated shopware

### LINKED ISSUE #244: Add error when not recent shopware version and not having SwagSecurity installed
### Please describe the feature you would like to see implemented.

_No response_

### PR DESCRIPTION
**Pull request by tinect**
Will move it also to main, if it's okay.

Closes #244

### CODE REVIEW

File: src/Components/Health/Checker/HealthChecker/SwagSecurityChecker.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        $application->setAutoExit(false);
+        $application->run($pluginRefresh, new NullOutput());
+    }
+
+    private function hasSecurityAdvisories(): bool
+    {
+        $cacheKey = \sprintf('security-advisories-%s', $this->shopwareVersion);
+        $cacheItem = $this->objectCache->getItem($cacheKey);
+
+        if ($cacheItem->isHit()) {
```
Comment (shyim) [inline src/Components/Health/Checker/HealthChecker/SwagSecurityChecker.php:87]:
```
$value = $cache->get('my_cache_key', function (ItemInterface $item): string {
    $item->expiresAfter(3600);

    // ... do some HTTP request or heavy computations
    $computedValue = 'foobar';

    return $computedValue;
});
```

looks better to read

https://symfony.com/doc/current/components/cache.html#cache-contracts

File: src/Components/Health/Checker/HealthChecker/SwagSecurityChecker.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        );
+    }
+
+    private function refreshPlugins(Connection $connection): void
+    {
+        $result = $connection->executeQuery(
+            'SELECT COUNT(*) FROM plugin WHERE updated_at < DATE_SUB(NOW(), INTERVAL 1 DAY)'
+        )->fetchOne();
+
+        if ($result === 0 || $result === '0') {
```
Comment (shyim) [inline src/Components/Health/Checker/HealthChecker/SwagSecurityChecker.php:70]: what is it now. this looks stupid then we can remove also one `=`

**Review by shyim — CHANGES_REQUESTED** [review]

File: src/Components/Health/Checker/HealthChecker/SwagSecurityChecker.php [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        #[Autowire(service: 'cache.object')]
+        private readonly AdapterInterface $objectCache
+    ) {}
+
+    public function collect(HealthCollection $collection): void
+    {
+        $this->refreshPlugins($this->connection);
+
+        try {
+            if ($this->hasSecurityAdvisories() === false) {
```
Comment (shyim) [inline src/Components/Health/Checker/HealthChecker/SwagSecurityChecker.php:34]:
```suggestion
            if (!$this->hasSecurityAdvisories()) {
```

File: src/Components/Health/Checker/HealthChecker/SwagSecurityChecker.php (hunk 1 above)
Comment (tinect) [reply src/Components/Health/Checker/HealthChecker/SwagSecurityChecker.php]: done :-)

File: src/Components/Health/Checker/HealthChecker/SwagSecurityChecker.php (hunk 2 above)
Comment (tinect) [reply src/Components/Health/Checker/HealthChecker/SwagSecurityChecker.php:71]: yeaaah, but if anyone reads with with one `=` someone could think this is a fault.

the used way it is more sure for that way.
But empty should be more save, isn't it?

**Review by shyim — APPROVED** [review]

**Pull request merged**